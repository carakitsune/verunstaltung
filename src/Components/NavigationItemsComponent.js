import React from 'react';
import Nav from 'react-bootstrap/Nav';
import { NavigationItemModel } from '../Models/NavigationItemModel';

export const navItems = [
    new NavigationItemModel('#home', 'Verunstaltungskalender'),
    new NavigationItemModel('#about', 'Über uns'),
    new NavigationItemModel('#past', 'Rückblick'),
    new NavigationItemModel('#contact', 'Kontakt')
]

export function NavigationItemsComponent() {
    var itemsJsx = navItems
    return itemsJsx.map((item, index) => 
        <Nav.Link key={index} href={item.href}>{item.label}</Nav.Link>
    );
}
