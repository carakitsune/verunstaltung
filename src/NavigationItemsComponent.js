import React from 'react';
import Nav from 'react-bootstrap/Nav';
import { NavigationItemModel } from './NavigationItemModel';

export const navItems = [
    new NavigationItemModel('#home', 'Verunstaltungskalender'),
    new NavigationItemModel('#about', 'Über uns'),
    new NavigationItemModel('#past', 'Rückblick'),
    new NavigationItemModel('#contact', 'Kontakt')
]

export function NavigationItemsComponent() {
    var itemsJsx = [];
    navItems.forEach(item => {
        itemsJsx.push(<Nav.Link href={item.href}>{item.label}</Nav.Link>);
    });
    return itemsJsx;
}
