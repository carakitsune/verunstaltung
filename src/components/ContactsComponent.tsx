import React from 'react';
import { Card } from 'react-bootstrap';
import { ContactModel } from '../models/ContactModel';

export const contacts = [new ContactModel('Siro Mancuso', 'mancusosiro.booking@gmail.com')];

export const ContactsComponent: React.FC = () => (
  <>
    {contacts.map((contact, index) => (
      <Card key={index} className="mt-4">
        <Card.Body>
          <Card.Title>{contact.name}</Card.Title>
          <Card.Link href={'mailto:' + contact.mail}>{contact.mail}</Card.Link>
        </Card.Body>
      </Card>
    ))}
  </>
);
