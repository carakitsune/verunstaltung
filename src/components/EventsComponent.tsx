import React from 'react';
import { Card } from 'react-bootstrap';
import { EventModel } from '../models/EventModel';
import img1 from './img1.png';
import img2 from './img2.png';

export const events = [new EventModel('Samstag 22.3.2020', 'Doors 20:00', [img1, img2])];

export const EventsComponent: React.FC = () => (
  <>
    {events.map((event, index) => (
      <Card key={index} className="mt-4">
        <Card.Body>
          <Card.Title className="date">
            {event.date} <br />
            {event.doors}
          </Card.Title>
          {event.pics.map((pic, key) => (
            <Card.Img key={key} className="mt-3" variant="top" src={pic} />
          ))}
        </Card.Body>
      </Card>
    ))}
  </>
);
