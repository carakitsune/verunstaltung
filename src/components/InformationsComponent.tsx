import React from 'react';
import { Card } from 'react-bootstrap';
import { InformationModel } from '../models/InformationModel';
import img3 from '../images/img3.png';

export const informationCards = [
  new InformationModel(
    'Die Verunstaltung',
    'Die Verunstaltung ist eine Veranstaltungsreihe, die im Zürcher El Lokal stattfindet. Bei den Verunstaltern \
            wird zusammen gehört, was nur so halb zusammengehört. Ziel ist es, das Publikum mit neuen Künstlern bekannt \
            zu machen, die wir Saite an Saite mit vertrauten Musikern präsentieren. Dabei gilt: Wieso in die Ferne \
            schweifen, wenn es so viele lokale talentierte Künstler gibt. Von Purismus halten wir uns aber rein. Das \
            einzige Klischee, das wir dabei bedienen ist, dass wir keine Klischees zu bedienen behaupten. Die Reihe ist \
            nicht ganz neu, sie ist bloss kürzlich erst erwachsen geworden: Noch unter dem Namen Seich’n’Delic wurde vor \
            rund vier Jahren eine Konzertreihe ins Leben gerufen, mit deren Erfolg auch das Bedürfnis nach einem breiter \
            gefächerten Programm und grösserer stilistischer Freiheit wuchs. Ein Namenswechsel war notwendig und die \
            Verunstaltung war geboren.',
    [img3],
  ),
  new InformationModel(
    'Die Veruntwortlichen',
    'Eure Verunstaltungsmacher, Siro Mancuso & Hans Frauchiger, das sind zwei Maturanden, die sich im Müssiggang \
        auf der schuleigenen Finnenbahn kennengelernt haben. Gegenwärtig geistern sie im relativen Raum des \
        Zwischenjahres herum, mit der Hoffnung, diese Herumfreigeisterei bringe möglichst viele brauchbare \
        Hirngespinster hervor. Die Verunstaltung ist eines der Hirngespinster, die derzeit in Erprobung sind. \
        Ehemals in Albisrieden geplant, wo die Umsetzung leider, leider kläglich an einem "Arschloch" gescheitert \
        ist, wird die Verunstaltung nun im renommierten El Lokal stattfinden.',
    [],
  ),
];

export const InformationsComponent: React.FC = () => (
  <>
    {informationCards.map((informationCard, index) => (
      <Card key={index} className="mt-4">
        <Card.Body>
          <Card.Title>{informationCard.title}</Card.Title>
          <Card.Text>{informationCard.text}</Card.Text>
          {informationCard.pics.map((pic, key) => (
            <Card.Img key={key} className="mt-3" variant="top" src={pic} />
          ))}
        </Card.Body>
      </Card>
    ))}
  </>
);
