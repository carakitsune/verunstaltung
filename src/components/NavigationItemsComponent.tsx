import React from 'react';
import Nav from 'react-bootstrap/Nav';
import { NavigationItemModel } from '../models/NavigationItemModel';

export const navItems = [
  new NavigationItemModel('#home', 'Verunstaltungskalender'),
  new NavigationItemModel('#about', 'Über uns'),
  new NavigationItemModel('#past', 'Rückblick'),
  new NavigationItemModel('#contact', 'Kontakt'),
];

export const NavigationItemsComponent: React.FC = () => (
  <>
    {navItems.map((item, index) => (
      <Nav.Link key={index} href={item.href}>
        {item.label}
      </Nav.Link>
    ))}
  </>
);
