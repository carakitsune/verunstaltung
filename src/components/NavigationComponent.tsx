import React from 'react';
import Navbar from 'react-bootstrap/Navbar';
import Nav from 'react-bootstrap/Nav';
import { NavigationItemsComponent } from './NavigationItemsComponent';

export const NavigationComponent: React.FC = () => (
  <Navbar collapseOnSelect expand={false} bg="white">
    <Navbar.Brand>Verunstaltungskalender</Navbar.Brand>
    <Navbar.Toggle aria-controls="responsive-navbar-nav" />
    <Navbar.Collapse id="responsive-navbar-nav">
      <Nav className="mr-auto">
        <NavigationItemsComponent />
      </Nav>
    </Navbar.Collapse>
  </Navbar>
);
