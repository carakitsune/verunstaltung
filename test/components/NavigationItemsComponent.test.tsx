import * as React from 'react';
import { NavigationItemsComponent } from '../../src/components/NavigationItemsComponent';
import { shallow } from 'enzyme';

describe('App', () => {
  it('renders NavigationComponent correctly', () => {
    expect(shallow(<NavigationItemsComponent />)).toMatchSnapshot();
  });
});
