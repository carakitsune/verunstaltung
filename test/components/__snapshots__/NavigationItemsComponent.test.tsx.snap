// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders NavigationComponent correctly 1`] = `
<NavLink
  as={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "displayName": "SafeAnchor",
      "render": [Function],
    }
  }
  disabled={false}
  href="#home"
>
  Verunstaltungskalender
</NavLink>
`;
